lib/SPIRV/SPIRVLowerMemmove.cpp

```diff
@@ -54,54 +54,51 @@ namespace SPIRV {
 
 class SPIRVLowerMemmoveBase {
 public:
-  SPIRVLowerMemmoveBase() : Context(nullptr), Mod(nullptr) {}
+  SPIRVLowerMemmoveBase() : Context(nullptr) {}
+
   void LowerMemMoveInst(MemMoveInst &I) {
+    // There is no direct equivalent of @llvm.memmove in SPIR-V and the closest
+    // instructions are 'OpMemoryCopy' and 'OpMemoryCopySized'.
+    //
+    // 'OpMemoryCopy' does not accept amount of bytes to copy and infers that
+    // from type which is being copied; also it only allows to copy value of a
+    // particular type to pointer pointing to the same type.
+    //
+    // 'OpMemoryCopySized' is closer to @llvm.memmove, because it actually
+    // copies bytes, but unlike memove it doesn't support overlapping source
```

Comment:
SPIR-V spec doesn't explicitly specify whether overlapping is supported or not. I think worth asking for clarification from Khronos,

Reply:
Good point, I've updated the comment in dbc54dd752c04bb65e8d2e2aedaaac22c247266e